src/main/scala/esmeta/analyzer/domain/value/TypeDomain.scala

```diff
@@ -505,7 +513,8 @@ object TypeDomain extends value.Domain {
   } yield p
 
   /** ast type check helper */
-  lazy val astDirectChildMap: Map[String, Set[String]] =
+  var astDirectChildMap: Map[String, Set[String]] = Map()
+  private def createAstDirectChildMap =
     (cfg.grammar.prods.map {
```

Comment:
Initializing `astDirectChildMap` with `createAstDirectChildMap` here fails because `cfg` in L518 refers to global analyzer.